gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -233,7 +234,7 @@ protected JobDetailImpl createJobDetailForReminderEvent(JobKey originalKey, JobK
       throws SchedulerException {
     JobDetailImpl jobDetail = (JobDetailImpl) this.schedulerService.getScheduler().getJobDetail(originalKey);
     jobDetail.setKey(reminderKey);
-    JobDataMap jobDataMap = jobDetail.getJobDataMap();
+    JobDataMap jobDataMap = deepCopyJobDataMap(jobDetail.getJobDataMap());
     jobDataMap = updatePropsInJobDataMap(jobDataMap, status, schedulerMaxBackoffMillis);
```

Comment:
I believe the copy should occur within `updatePropsInJobDataMap`.  specifically, try calling:
```
Properties copyOfPrevJobProps = (Properties) ((Properties) jobDataMap.get(GobblinServiceJobScheduler.PROPERTIES_KEY)).clone();
```

Reply:
clone creates a shallow copy of the Properties object. It seems safer to provide a deep copy of the jobDataMap (and Properties obj) to be copied which can be modified in the update method

Reply:
sure - in addition to `.clone()`, you must also recreate the `jobDataMap`, so the new JDM references the cloned props, but the old one does not.

but, why not encapsulate that within `updatePropsInJobDataMap` (potentially renamed)?  it already returns a `JobDataMap` result.  just specify in its contract that will be a copy with the input arg NOT modified.  (e.g. `copyOnUpdateJobDataMapProps`)

is there some reason elsewhere to continue with in-place JDM update?  as we've seen here, such mods have the potential to introduce subtle errors...